includes/kco-functions.php

```diff
@@ -680,6 +680,51 @@ function kco_validate_order_total( $klarna_order, $order ) {
 	return true;
 }
 
+/**
+ * Validate the Klarna Checkout item quantities against the Woo order.
+ *
+ * @param array    $klarna_order The Klarna order.
+ * @param WC_Order $order The Woo order.
+ *
+ * @return bool
+ */
+function kco_validate_cart_content( $klarna_order, $order ) {
+	foreach ( $order->get_items() as $item ) {
+		// Only check quantity for product items.
+		if ( 'line_item' !== $item->get_type() ) {
+			continue;
+		}
+
+		// Assume the item is a mismatch until proven otherwise. This way, in the unlikely chance of not finding any matching item, we can set the order to on-hold.
+		$mismatch = true;
+
+		// Try the specific (variation id) before the general (product id).
+		$product_id = 0 !== $item->get_variation_id() ? $item->get_variation_id() : $item->get_product_id();
+		$product    = wc_get_product( $product_id );
+		$sku        = strval( $product->get_sku() ?? $product->get_id() );
+		foreach ( $klarna_order['order_lines'] as $klarna_order_item ) {
+			if ( $klarna_order_item['reference'] === $sku ) {
+				$mismatch = $klarna_order_item['quantity'] !== $item->get_quantity();
+			}
+		}
```

Comment:
This does not include a check to ensure that Klarna and WooCommerce have the same order lines, only that the klarna order lines that has a matching sku in the WooCommerce order has the same quantity.

We should be checking that the order lines match, so every order line in the WooCommerce order should exist in Klarna, and all order lines that exists in Klarna should exist in WooCommerce. But keep in mind the rounding line we add in some cases, which is not added to the WooCommerce order.

This also does not have any validation for the shipping lines either, which we should add in, just make sure that we check properly for if shipping is in the iframe or not.